### PR TITLE
Fixed rubocop and foodcritic for recipes before chefspec tests

### PR DESCRIPTION
**Pull request by cthalmann**
Chefspec tests need to be written, but before it will even get to those in rake, this cookbook needs to pass rubocop and foodcritic. With this pull request, it will do those things. I also deleted linter_spec.rb since that does not match our current set up (and also was causing rubocop errors) and will have to be rewritten anyway.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -24,7 +24,7 @@
   source '60-haproxy.conf.erb'
   owner 'root'
   group 'root'
-  mode 0644
+  mode 0o644
```
Comment (ramereth) [inline recipes/default.rb]: Please add the following to ``.rubocop.yml``:

```
Style/NumericLiteralPrefix:
  EnforcedOctalStyle: zero_only
```

Then revert this changed that rubocop made. We aren't using this as a standard currently.

File: spec/linter_spec.rb [hunk 2]
```diff
@@ -1,78 +0,0 @@
...
-  end
-
-  it 'should not raise an error with correct config' do
-    linter = OSLHAProxyLinter.new
-    good_array = %w(testing1 testing2)
-    expect do
-      linter.check_haproxy_config(good_array)
-    end.not_to raise_error
-  end
-end
```
Comment (ramereth) [inline spec/linter_spec.rb]: Why is this being removed entirely? I know this isn't ChefSpec but raw Rspec and I think it should stay.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,14 +1,29 @@
-$LOAD_PATH << File.join(File.dirname(__FILE__), '..')
```
Comment (ramereth) [inline spec/spec_helper.rb]: We probably should retain this spec_helper code for the other file, but probably put it in a different file form our normal chefspec helper file. Perhaps put the contents of it in a file called ``rspec_helper.rb`` and have this be the updated version we normally use.

File: recipes/data_bag.rb [hunk 4]
```diff
@@ -43,7 +43,7 @@
 # Create a linter for ensuring a valid config. Defined in libraries/
 linter = OSLHAProxyLinter.new
 
-config.each do |project|
+config.each do |project| # rubocop:disable Metrics/BlockLength
```
Comment (ramereth) [inline recipes/data_bag.rb]: I get a rubocop warning (ironically) here.

File: spec/rspec_helper.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+$LOAD_PATH << File.join(File.dirname(__FILE__), '..')
+
+require 'bundler/setup'
+Bundler.setup
```
Comment (ramereth) [inline spec/rspec_helper.rb]: Please remove this and the line above it as this is causing issues with our standard ruby env with chefdk.

**Comment by ramereth** [comment]
You also need to change the following line from:

``` ruby
require 'libraries/osl_haproxy_linter'
```

to:
``` ruby
require_relative '../libraries/osl_haproxy_linter'
```

**Comment by ramereth** [comment]
@cthalmann mind rebasing into a single commit please?

**Comment by cthalmann** [comment]
!bump patch ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v2.3.1.  Have a nice day!

**Pull request merged**